arelle/RenderingLayout-draft.py

```diff
@@ -198,55 +143,36 @@ def childContainsOpenNodes(childStructuralNode):
                 return True
         return False
 
+    
+def checkLabelWidth(view, strctMdlNode, subtreeRels, checkBoundFact=False):
+    if strctMdlNode.axis == "y":
+        # messages can't be evaluated, just use the text portion of format string
+        label = strctMdlNode.header(lang=view.lang, 
+                                      returnGenLabel=not checkBoundFact, 
+                                      returnMsgFormatString=not checkBoundFact)
+        if label:
+            # need to et more exact word length in screen units
+            widestWordLen = max(len(w) * RENDER_UNITS_PER_CHAR for w in label.split())
+            # abstract only pertains to subtree of closed nodesbut not cartesian products or open nodes
+            while strctMdlNode.depth >= len(view.rowHdrColWidth):
+                view.rowHdrColWidth.append(0)
```

Comment:
appending a `0` seems odd, but suggest `view.rowHdrColWidth.extend(0 for _ in range(len(view.rowHdrColWidth)-structMdlNode.depth)`